tests/unit/test_websocket_router.py

```
@@ -0,0 +1,372 @@
+class TestWebSocketMessageLoop:
+    """Tests for WebSocket message loop."""
+
+    @pytest.mark.asyncio
+    async def test_message_loop_handles_connection_error(self) -> None:
+        """Test message loop exits on connection error during ping."""
+        mock_websocket = AsyncMock()
+
+        # Use a side effect that raises TimeoutError then ConnectionError
+        call_count = 0
+
+        async def mock_wait_for(*args, **kwargs):
```

Ruff (0.14.8)

Missing return type annotation for private function `mock_wait_for`

Add return type annotation: `NoReturn`

(ANN202)